### PR TITLE
end_of_line

### LINKED ISSUE #1: `end_of_line` property
http://editorconfig.org
> end_of_line: set to lf, cr, or crlf to control how line breaks are represented.

I couldn't figure out a way to change this. Suggestion/PR welcome.

### PR DESCRIPTION
**Pull request by milan**
Master branch of Atom includes atom/text-buffer 6.6.1 which supports changing of line endings. Won't work will the released version of atom yet, but just doing this while I have the time.

-

Fixes #1

### CODE REVIEW

**Comment by milan** [comment]
#1 Is being fixed as part of Atom's Quality Initiative

File: index.js [hunk 1]
```diff
@@ -31,6 +37,13 @@ function init(editor) {
 			}
 		}
 
+		if (config.end_of_line && lineEndings.hasOwnProperty(config.end_of_line)) {
+			var preferredLineEnding = lineEndings[config.end_of_line];
+			var buffer = editor.getBuffer();
+			buffer.setPreferredLineEnding(preferredLineEnding);
```
Comment (sindresorhus) [inline index.js]: Kinda disappointing the API won't change the existing text for us.

File: index.js [hunk 2]
```diff
@@ -31,6 +37,13 @@ function init(editor) {
...
+			buffer.setText(buffer.getText().replace(/\n|\r/g, preferredLineEnding));
```
Comment (sindresorhus) [inline index.js]: The regex is faulty, should be `/\r?\n|\r/g`.

File: index.js [hunk 3]
```diff
@@ -8,6 +8,12 @@ function init(editor) {
 
 	const file = editor.getURI();
 
+	const lineEndings = {
+		crlf: "\r\n",
+		lf: "\n",
+		cr: "\r"
```
Comment (sindresorhus) [inline index.js]: single-quotes

**Comment by sindresorhus** [comment]
Awesome! Glad to see this being implemented :)

File: index.js (hunk 1 above)
Comment (milan) [inline index.js]: The atom/line-ending-selector plugin pretty much does the same thing, so not sure they've implemented it the way you'd prefer.

**Comment by sindresorhus** [comment]
Would you mind adding support for this in the new generate command too?
https://github.com/sindresorhus/atom-editorconfig/blob/368c452b85d1888b2e896de11cbaa62ebf265876/commands/generate.js#L18

And update the mininum Atom version to 1.0.8 (which I assume will be next one with support for this).

**Comment by milan** [comment]
Updated the version, but wasn't too sure what you meant by updated the generate command (apart from merging it in) as it looks like it should generate a correct end_of_line value.

**Comment by milan** [comment]
Actually I see what you mean, I think the config will be called editor.preferredLineEnding, but would prefer to wait for 1.0.8 to be released as that is a bit of a guess.

**Comment by sindresorhus** [comment]
Seems to be released now: https://github.com/atom/atom/releases/tag/v1.0.8

**Comment by sindresorhus** [comment]
Thanks @milan :) Published as 1.2.0.

**Comment by florianb** [comment]
:+1:

**Pull request closed without merging**